src/main/java/com/onarandombox/MultiverseCore/anchor/AnchorManager.java

```diff
@@ -140,18 +140,15 @@ public Set<String> getAnchors(Player p) {
             if (ancLoc == null) {
                 continue;
             }
-            String worldPerm = "multiverse.access." + ancLoc.getWorld().getName();
-            // Add to the list if we're not enforcing access
-            // OR
-            // We are enforcing access and the user has the permission.
-            if (!this.plugin.getMVConfig().getEnforceAccess() ||
-                    (this.plugin.getMVConfig().getEnforceAccess() && p.hasPermission(worldPerm))) {
-                myAnchors.add(anchor);
-            } else {
-                Logging.finer(String.format("Not adding anchor %s to the list, user %s doesn't have the %s " +
-                        "permission and 'enforceaccess' is enabled!",
-                        anchor, p.getName(), worldPerm));
+            MVWorld mvWorld = this.plugin.getMVWorldManager().getMVWorld(ancLoc.getWorld());
+            if (mvWorld != null) {
+                if (!this.plugin.getPlayerActionChecker().hasAccessToWorld(p, mvWorld).isSuccessful()) {
+                    Logging.finer("Player '%s' does not have access to anchor '%s' in world '%s'.",
+                            p.getName(), anchor, mvWorld.getName());
+                    continue;
+                }
```

Comment:
It would be nice to extract this out to a new method to enhance readability.

Reply:
Yea, but mainly bcu I want to revamp anchors anyways. I want to create an anchor object which contains the anchor name and location.